argparser.cpp

```diff
@@ -0,0 +1,129 @@
+#include <Windows.h>
+#include <tchar.h>
+#include "cpp_tstring.hpp"
+#include <iterator>
+#include <vector>
+#include "argparser.hpp"
+
+TCHAR helpMessage[] = TEXT("Usage: [program] [--output file] [--enable_cache cache_file] [--source source]\n\
```

Comment:
Why is the help message not marked `constexpr`? Having a text this large in a writeable memory page may impact performance, in case the page needs an extra copy-on-write.